projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_router.py

```diff
@@ -59,6 +70,12 @@ def send_object_for_ingestion(
         target: Optional[str] = None,
         collection_id: Optional[str] = None,
     ):
+        # Initialize the pre- and post- processors.
```

Comment:
filter by lambda, then check if processors could match the configured, else raise an exception;
take a loot at `__initialize_ingester` to reuse some parts of it?maybe simplify it to apply to all cases? could lets sync on this one to elaborate on options if needed

Reply:
Refactored to be cleaner.